projects/aca-content/folder-rules/src/manage-rules/manage-rules.smart-component.spec.ts

```diff
@@ -224,88 +246,205 @@ describe('ManageRulesSmartComponent', () => {
+    const ruleDetails = unitTestingUtils.getByCSS('aca-rule-details');
 
-    deleteRuleBtn.click();
+    component.onRuleDeleteButtonClicked(ruleMock('owned-rule-1'));
 
     fixture.detectChanges();
     folderRulesService.deletedRuleId$ = of('owned-rule-1-id');
 
-    expect(onRuleDeleteButtonClickedSpy).toHaveBeenCalled();
-    expect(dialogOpenSpy).toHaveBeenCalled();
+    expect(dialog.open).toHaveBeenCalled();
     expect(deleteRuleSpy).toHaveBeenCalled();
-    expect(onRuleDeleteSpy).toHaveBeenCalledTimes(1);
     expect(rules).toBeTruthy('expected rules');
     expect(ruleDetails).toBeTruthy('expected ruleDetails');
-    expect(deleteRuleBtn).toBeTruthy();
+  });
+
+  it('should refresh main rule set when link rules dialog is closed', () => {
+    folderRuleSetsService.folderInfo$ = of(owningFolderMock);
+    folderRuleSetsService.mainRuleSet$ = of(ownedRuleSetMock);
+    folderRuleSetsService.inheritedRuleSets$ = of([inheritedRuleSetMock]);
+    folderRuleSetsService.isLoading$ = of(false);
+    folderRulesService.selectedRule$ = of(ruleMock('owned-rule-1'));
+    folderRulesService.deletedRuleId$ = of(null);
+    actionsService.loading$ = of(false);
+
+    fixture.detectChanges();
+
+    spyOn(dialog, 'open').and.returnValue({
+      afterClosed: () => of(true)
+    } as MatDialogRef<boolean>);
+    const refreshMainRuleSetSpy = spyOn(folderRuleSetsService, 'refreshMainRuleSet');
+
+    fixture.detectChanges();
+    expect(component).toBeTruthy('expected component');
+
+    const rules = unitTestingUtils.getAllByCSS('.aca-rule-list-item');
+    const ruleDetails = unitTestingUtils.getByCSS('aca-rule-details');
+
+    component.openLinkRulesDialog();
+
+    fixture.detectChanges();
+    folderRulesService.deletedRuleId$ = of('owned-rule-1-id');
+
+    expect(dialog.open).toHaveBeenCalled();
+    expect(refreshMainRuleSetSpy).toHaveBeenCalled();
+    expect(rules).toBeTruthy('expected rules');
+    expect(ruleDetails).toBeTruthy('expected ruleDetails');
+  });
+
+  it('should call deleteRuleSetLink when onRuleSetUnlinkClicked is called', () => {
+    folderRuleSetsService.folderInfo$ = of(owningFolderMock);
+    folderRuleSetsService.mainRuleSet$ = of(ownedRuleSetMock);
+    folderRuleSetsService.inheritedRuleSets$ = of([inheritedRuleSetMock]);
+    folderRuleSetsService.isLoading$ = of(false);
+    folderRulesService.selectedRule$ = of(ruleMock('owned-rule-1'));
+    folderRulesService.deletedRuleId$ = of(null);
+    actionsService.loading$ = of(false);
+
+    fixture.detectChanges();
+
+    spyOn(dialog, 'open').and.returnValue({
+      afterClosed: () => of(true)
+    } as MatDialogRef<boolean>);
+    const deleteRuleSetLinkSpy = spyOn(folderRuleSetsService, 'deleteRuleSetLink');
+
+    fixture.detectChanges();
+    expect(component).toBeTruthy('expected component');
+
+    const rules = unitTestingUtils.getAllByCSS('.aca-rule-list-item');
+    const ruleDetails = unitTestingUtils.getByCSS('aca-rule-details');
```

Comment:
aca-rule-details is selector of component so please use getByDirective function instead to get that element